TemplateApplication/Schedule/TemplateApplicationScheduler.swift

```diff
@@ -7,43 +7,41 @@
 //
 
 import Foundation
+import Spezi
 import SpeziScheduler
+import class ModelsR4.Questionnaire
+import class ModelsR4.QuestionnaireResponse
 
 
-/// A `Scheduler` using the ``TemplateApplicationTaskContext`` to schedule and manage tasks and events in the
-/// Spezi Template Application.
-typealias TemplateApplicationScheduler = Scheduler<TemplateApplicationTaskContext>
+final class TemplateApplicationScheduler: Module, DefaultInitializable {
+    @Dependency(Scheduler.self) private var scheduler
 
-
-extension TemplateApplicationScheduler {
-    static var socialSupportTask: SpeziScheduler.Task<TemplateApplicationTaskContext> {
-        let dateComponents: DateComponents
-        if FeatureFlags.testSchedule {
-            // Adds a task at the current time for UI testing if the `--testSchedule` feature flag is set
-            dateComponents = DateComponents(
-                hour: Calendar.current.component(.hour, from: .now),
-                minute: Calendar.current.component(.minute, from: .now)
-            )
-        } else {
-            // For the normal app usage, we schedule the task for every day at 8:00 AM
-            dateComponents = DateComponents(hour: 8, minute: 0)
+    init() {}
+    
+    /// Add or update the current list of task upon app startup.
+    func configure() {
+        do {
+            // TODO: support test schedule flag: FeatureFlags.testSchedule for UI testing!
+            try scheduler.createOrUpdateTask(
+                id: "social-support-questionnaire",
+                title: "TASK_SOCIAL_SUPPORT_QUESTIONNAIRE_TITLE",
+                instructions: "TASK_SOCIAL_SUPPORT_QUESTIONNAIRE_DESCRIPTION",
+                schedule: .daily(hour: 8, minute: 0, startingAt: .today)
+            ) { context in
+                context.questionnaire = Bundle.main.questionnaire(withName: "SocialSupportQuestionnaire")
+            }
+        } catch {
+            // TODO: we should visualize this? or at least allow to visualize it!
```

Comment:
Maybe a view state that is exposed to the UI and could be used to tigger a view site alert?

Reply:
We added a viewState that is shown as an alert.